test/extended/util/disruption/frontends/known_backends.go

```diff
@@ -4,55 +4,92 @@ import (
 	"context"
 
 	configclient "github.com/openshift/client-go/config/clientset/versioned"
+	routeclient "github.com/openshift/client-go/route/clientset/versioned"
 	"github.com/openshift/origin/pkg/monitor"
 	"github.com/openshift/origin/pkg/monitor/backenddisruption"
+	exutil "github.com/openshift/origin/test/extended/util"
 	"github.com/openshift/origin/test/extended/util/cluster"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
 	"k8s.io/client-go/rest"
 	e2e "k8s.io/kubernetes/test/e2e/framework"
 )
 
+const (
+	oauthRouteNamespace = "openshift-authentication"
+	oauthRouteName      = "oauth-openshift"
+)
+
 func StartAllIngressMonitoring(ctx context.Context, m monitor.Recorder, clusterConfig *rest.Config) error {
-	if err := createOAuthRouteAvailableWithNewConnections().StartEndpointMonitoring(ctx, m, nil); err != nil {
+	// Ingress monitoring checks for oauth and console routes to monitor healthz endpoints. Check availability
+	// before setting up any monitors.
+	routeAvailable, err := isRouteAvailable(ctx, clusterConfig, oauthRouteNamespace, oauthRouteName)
+	if err != nil {
 		return err
 	}
-	if err := createOAuthRouteAvailableWithConnectionReuse().StartEndpointMonitoring(ctx, m, nil); err != nil {
-		return err
+	if routeAvailable {
```

Comment:
Is there any OpenShift installation which does not deploy routes? Given routes are deployed by default and MicroShift deploys them as well, why to perform the check?

Reply:
There are two routes in OCP, which come from the oauth api server and the console operator. The former is always installed as it is a core operator. The latter can be disabled in 4.12+.
Neither of those exist in MicroShift, so the routes wont exist unless they are introduced in the future. In an OCP deployment there should always be (at least) a route for the oauth api server.